### PR TITLE
Tweak cluster behavior so it tries other nodes in case a node goes down

### PR DESCRIPTION
**Pull request by myarichuk**
In clusters, for ReadFromAllXXX failover behaviors, make sure to try other cluster nodes before failing

### CODE REVIEW

File: Raven.Client.Lightweight/Connection/Request/ClusterAwareRequestExecuter.cs [hunk 1]
```diff
@@ -310,17 +312,46 @@ private OperationMetadata GetNodeForReadOperation(OperationMetadata node)
             var nodes = new List<OperationMetadata>(NodeUrls);
 
             if (readStripingBase == -1)
-                return LeaderNode;
+            {
+                var nodeForReadOperation = LeaderNode;
+                if (ShouldExecuteUsing(nodeForReadOperation))
+                    return nodeForReadOperation;
+                
+                throw new InvalidOperationException("Leader node seem to be down down, and we are under ForceReadFromMaster() scope.");
```
Comment (ayende) [inline Raven.Client.Lightweight/Connection/Request/ClusterAwareRequestExecuter.cs]: Better to make the request, then fail it without additional retries.

File: Raven.Client.Lightweight/Connection/Request/ClusterAwareRequestExecuter.cs [hunk 2]
```diff
@@ -310,17 +312,46 @@ private OperationMetadata GetNodeForReadOperation(OperationMetadata node)
...
+            }
 
             if (nodes.Count == 0)
-                return null;
+            {
+                if (ShouldExecuteUsing(node))
+                    return node;
 
+                throw new InvalidOperationException("Topology nodes list is empty, and the primary node seem to be down (more than one failure to connect). ");
```
Comment (ayende) [inline Raven.Client.Lightweight/Connection/Request/ClusterAwareRequestExecuter.cs]: Same, make the request so they'll get the actual error

File: Raven.Client.Lightweight/Connection/Request/ClusterAwareRequestExecuter.cs [hunk 3]
```diff
@@ -310,17 +312,46 @@ private OperationMetadata GetNodeForReadOperation(OperationMetadata node)
...
+                if (ShouldExecuteUsing(readNode))
+                {
+                    foundAvailableNode = true;
+                    break;
+                }
+                faultedNodes++;
+            } while(faultedNodes < nodes.Count - 1); //go over all nodes except the initial
+
+            if(foundAvailableNode == false)
+                throw new InvalidOperationException("Cluster is not reachable. Executing operation on any of the nodes failed, aborting.");
```
Comment (ayende) [inline Raven.Client.Lightweight/Connection/Request/ClusterAwareRequestExecuter.cs]: Same

**Comment by myarichuk** [comment]
updated PR according to the review

**Comment by myarichuk** [comment]
Close and reopen to run the tests again

**Pull request closed without merging**